### PR TITLE
Added Tags filter for budgets and updated the api version

### PR DESCRIPTION
**Pull request by asarkar84**
Added Tags filter for budgets and updated the api version

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by sandeepnl — APPROVED** [review]

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (marstr) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Looks like this is a  duplicate of this parameter: https://github.com/Azure/azure-rest-api-specs/pull/2625/files#diff-dca9b11ab859f276378213344e091b34R342

Moving the definition into the "defintions" section and referencing it both places could improve the maintainability of this document moving forward.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (marstr) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Out of curiosity, must this be a GUID or anything else that could be hinted to with a "format"

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/BudgetsList.json
Comment (marstr) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/BudgetsList.json]: Looks like evidence of the comment above, that we could hint at the format of `meterId`s.

Also, for what it's worth, I recommend stripping or obfuscating GUIDs so that nothing specific to your service is available in the configs.

File: specification/consumption/resource-manager/readme.md
Comment (marstr) [inline specification/consumption/resource-manager/readme.md]: Is there still a January version of this, or did shipping schedule change?

File: specification/consumption/resource-manager/readme.md
Comment (marstr) [inline specification/consumption/resource-manager/readme.md]: Depending on your response to my comment:

> Is there still a January version of this, or did shipping schedule change?

The `package-2018-01 and go` clause may need to be deleted.

**Review by marstr — CHANGES_REQUESTED** [review]
Overall, an excellent looking Swagger!

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: reservationOrderIdParameter and reservationIdParameter are two different parameters which are going to be part of the URI so we have defined them as Parameters. 

Please let me now if I missed the point.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sure! I have added the hint saying it is a GUID.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/BudgetsList.json
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/BudgetsList.json]: Very good point. I have removed the actual GUIDs from all examples and added default GUID.

File: specification/consumption/resource-manager/readme.md
Comment (asarkar84) [reply specification/consumption/resource-manager/readme.md]: Good catch! It was just overwritten. I have added the January version back.

File: specification/consumption/resource-manager/readme.md
Comment (asarkar84) [reply specification/consumption/resource-manager/readme.md]: Good catch! It was just overwritten. I have added the January version back.

**Review by asarkar84 — COMMENTED** [review]
Thank you! for the thorough review. I have incorporated  the review comments.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (marstr) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Ah sorry, it looks like my comment is based on the same line as it was before or something. I'm speaking to "grain" and "datagrain" enums.
 
Unless there's a subtle difference that I'm just overlooking, It seems that the following definition appears multiple times in this PR:

``` JSON
          {
            "name": "grain",
            "description": "Can be daily or monthly",
            "in": "query",
            "required": true,
            "type": "string",
            "enum": [
              "daily",
              "monthly"
            ],
            "x-ms-enum": {
              "name": "datagrain",
              "modelAsString": true,
              "values": [
                {
                  "value": "daily",
                  "description": "Daily grain of data",
                  "name": "DailyGrain"
                },
                {
                  "value": "monthly",
                  "description": "Monthly grain of data",
                  "name": "MonthlyGrain"
                }
              ]
            }
          }
```

**Comment by marstr** [comment]
I replied to the duplication comment to make more clear what I meant. It's probably not a blocking issue, but having only one definition of an enum will make the generated code a little more usable. :)

**Comment by marstr** [comment]
Oh another heads up, I'm going to be OOF for just over two weeks starting Friday. If @ravbhatnagar hasn't signed off by then I'll have to reassign this PR.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Thank you! for sharing more details. I have made the necessary changes and updated the PR.

**Comment by asarkar84** [comment]
I have made necessary changes for the last review comment as well. Please take a look.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
This looks good from the SDK perspective, we'll still need to wait for @ravbhatnagar or another ARM representative to sign-off before we can merge.

**Comment by ravbhatnagar** [comment]
Looks good.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (marstr) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sorry I didn't notice this earlier! Adding a hint to the description is an improvement, but I actually mean that you could add a "format" portion of this definition, set to "uuid".

As [the relevant part of the Swagger spec says](https://swagger.io/specification/#dataTypeFormat):
>  the format property is an open string-valued property, and can have any value. Formats such as "email", "uuid", and so on, MAY be used even though undefined by this specification. 


Which would mean this is what these blocks would look like:

``` JSON
"meterId":{
    "description": "The meter id (GUID).",
    "type":"string",
    "format":"uuid",
    "readOnly":true
},
```

**Review by marstr — CHANGES_REQUESTED** [review]
Noticed one more quick thing!

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sure! I have incorporated this review comment as well.

**Review by marstr — APPROVED** [review]

**Pull request merged**
